File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string s1 = "";
        for(int i = 0; i<s.size(); i++){
            char ch = s[i];
            if(ch>='a' && ch<='z'){
                s1+=ch;
            }
            else if(ch>='A' && ch<='Z'){
                s1+=tolower(ch);
            }else if(ch>='0' && ch<='9'){
                s1+=ch;
            }
        }
        string s2 = s1;
        reverse(s1.begin(), s1.end());
        if(s1==s2){
            return true;
        }
        return false;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int  l = 0;
        int n = s.length();
        String s1 = "";
        for(int i = 0; i<n; i++){
            char ch = s.charAt(i);
            if(Character.isLetterOrDigit(ch)){
                s1+=Character.toLowerCase(ch);
            }
        }
        int r = s1.length()-1;
        while(l<r){
            if(s1.charAt(l)!=s1.charAt(r)){
                return false;
            }
            l++;
            r--;
        }
        return true;
    }
}